mongo/tests/test_integration.py

```diff
@@ -586,6 +586,48 @@ def test_db_names_missing_existent_database(check, instance_integration, aggrega
     assert len(aggregator._events) == 0
 
 
+@auth
+@pytest.mark.usefixtures('dd_environment')
+def test_mongod_auth_ok(check, dd_run_check, aggregator):
+    instance = {
+        'hosts': ['{}:{}'.format(HOST, PORT1)],
+        'username': 'testUser',
+        'password': 'testPass',
+        'options': {'authSource': 'authDB'},
+    }
+    mongo_check = check(instance)
+    dd_run_check(mongo_check)
+    aggregator.assert_service_check('mongodb.can_connect', status=MongoDb.OK)
+
+
+@auth
+@pytest.mark.usefixtures('dd_environment')
+def test_mongod_auth_bad_user(check, dd_run_check, aggregator):
+    instance = {
+        'hosts': ['{}:{}'.format(HOST, PORT1)],
+        'username': 'badUser',
+        'password': 'testPass',
+        'options': {'authSource': 'authDB'},
+    }
+    with pytest.raises(Exception, match="Authentication failed"):
+        mongo_check = check(instance)
+        dd_run_check(mongo_check)
+
+
+@auth
+@pytest.mark.usefixtures('dd_environment')
+def test_mongod_auth_bad_password(check, dd_run_check, aggregator):
```

Comment:
I think you can parametrize `test_mongod_auth_bad_password()` and `test_mongod_auth_bad_user()` into the same test function since you are expecting the same error. Example: https://github.com/DataDog/integrations-core/blob/709a9d0aef374a82db98211a5bdbac57ea5d708b/ibm_mq/tests/test_ibm_mq_unit.py#L116-L125

Reply:
Good point! Thx!